File: api/src/features/credentials/dto/credential_response.rs
use crate::{features::credentials::CredentialStatus, *};

#[derive(Debug, Clone, Serialize, Deserialize, OperationIo, JsonSchema, Default)]
pub struct CredentialResponse {
    #[schemars(description = "ID of the credential")]
    pub pk: Partition,

    #[schemars(description = "Name of the credential")]
    pub name: String,

    #[schemars(description = "API key prefix for display")]
    pub api_key_prefix: String,

    #[schemars(description = "Status of the credential")]
    pub status: CredentialStatus,

    #[schemars(description = "Creation timestamp")]
    pub created_at: i64,

    #[schemars(description = "Last used timestamp")]
    pub last_used_at: Option<i64>,

    #[schemars(description = "The full API key (only returned on creation)")]
    pub api_key: Option<String>,
}

File: web/src/features/credentials/dto/credential-response.ts
import { CredentialStatus } from "../types/credential-status";

export interface CredentialResponse {
  id: string;
  name: string;
  api_key_prefix: string;
  status: CredentialStatus;
  created_at: number;
  last_used_at?: number;
  api_key?: string; // Only present on creation
}
